scripts/locations_data_load/locations_lambda.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python3
+
+import requests
+
+import pandas as pd
+import boto3
+import uuid
+import datetime
+from boto3.dynamodb.conditions import Attr
+
+
+def lambda_handler(event, context):
+    print("Fetching organizations data.")
+    fetch_organizations()
+    print("Fetching Y organizations data.")
+    fetch_Y_organizations()
+
+
+# Get parameters from store
+def get_ssm(id, secret):
+    ssm = boto3.client("ssm")
+    client_id = ssm.get_parameter(Name=id)["Parameter"]["Value"]
+    client_secret = ssm.get_parameter(Name=secret, WithDecryption=True)["Parameter"][
+        "Value"
+    ]
+    return client_id, client_secret
+
+
+def get_api_token():
+    ssm_param_id = "/data/api/lambda/client_id"
+    ssm_param_sec = "/data/api/lambda/client_secret"
+    headers = {
+        "Content-Type": "application/x-www-form-urlencoded",
+        "Accept": "*/*",
+        "Accept-Encoding": "gzip, deflate, br",
+        "Connection": "Keep-alive",
+    }
+
+    data = {
+        "grant_type": "client_credentials",
+        "client_id": get_ssm(ssm_param_id, ssm_param_sec)[0],
+        "client_secret": get_ssm(ssm_param_id, ssm_param_sec)[1],
+    }
+
+    response = requests.post(url=token_api_endpoint, headers=headers, data=data)
+    token = response.json().get("access_token")
+
+    return token
+
+
+def get_headers():
+    token = get_api_token()
+    headers = {"Authorization": "Bearer " + token}
+    return headers
+
+
+def read_ods_api(api_endpoint, headers, params):
+    try:
+        response = requests.get(api_endpoint, headers=headers, params=params)
+
+        # Check if the request was successful (status code 200)
+        if response.status_code == 200:
+            response_data = response.json()
+            return response_data
+        else:
+            print(f"Error: {response.status_code} - {response.text}")
+            return None
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        return None
+
+
+def generate_random_id():
+    # Generate a random 16-digit ID
+    return str(uuid.uuid4().int)[0:16]
+
+
+# function to get the current date and time in UK format
+def get_formatted_datetime():
+    current_datetime = datetime.datetime.now()
+    return current_datetime.strftime("%d-%m-%Y %H:%M:%S")
+
+
+def capitalize_line(line):
+    return line.title()
+
+
+def capitalize_address_item(address_item):
+    capitalized_item = {}
+
+    for key, value in address_item.items():
+        if key == "line" and isinstance(value, list):
+            capitalized_item[key] = [capitalize_line(line) for line in value]
+        elif key in ["city", "district", "country"]:
+            capitalized_item[key] = value.title()
+        elif key == "postalCode":
+            capitalized_item[key] = value
+        elif key != "extension":
```

Comment:
Sonarcloud is reporting this as duplicate of line above so maybe we could combine the two ?